utils/SPIRVRunner/SPIRVRunner.cpp

```
@@ -7,8 +7,152 @@
+using ordered_json = nlohmann::ordered_json;
+
+// Structure that contains Triton kernel arguments
+struct argsDict {
+    int gridX;
+    int gridY;
+    int gridZ;
+    int num_ctas;
+    int num_stages;
+    int num_warps;
+    int threads_per_warp;
+    int shared_memory;
+    std::string kernel_name;
+    std::string spv_name;
+    std::vector<torch::Tensor> tensor_vec;
+    std::vector<std::string> ttype_vec;
+    std::vector<int> tensor_iarg_vec;
+    std::tuple<int, std::string> outTensorProp;
+    ordered_json jsonData;
+
+    argsDict(int gX =0, int gY = 0, int gZ = 0, int ctas = 0,
+            int stages = 0, int warps = 0, int tpw = 0,int sm = 0) 
+            : gridX(gX), gridY(gY), gridZ(gZ), num_ctas(ctas), num_stages(stages),
+            num_warps(warps), threads_per_warp(tpw), shared_memory(sm) {}
+    
+    void addTensor(const torch::Tensor& tensor) {
+        tensor_vec.push_back(tensor);
+    }
+
+    void addTensorType(std::string type) {
+        ttype_vec.push_back(type);
+    }
+
+    void addTensorIarg(int arg) {
+        tensor_iarg_vec.push_back(arg);
+    }
+};
+
+// Function to extract the numerical part from keys like "tensor_9"
+int extractNumber(const std::string& key) {
+    std::regex number_pattern(R"(\d+)");
+    std::smatch match;
+    if (std::regex_search(key, match, number_pattern)) {
+        return std::stoi(match.str());
+    }
+    return 0;
+}
+
+auto load_tensor(const std::string &filename) {
+    std::ifstream ins(filename, std::ios::binary);
+    if (!ins.is_open()) {
+      throw std::runtime_error("Failed to open file " + filename);
+    }
+    std::vector<char> buffer((std::istreambuf_iterator<char>(ins)), std::istreambuf_iterator<char>());
+    try {
+        auto ivalue = torch::pickle_load(buffer);
+        if (ivalue.isTensor()) {
+          auto tensor = ivalue.toTensor();
+          return tensor;
+        } else { 
+          std::cerr << "Error: Loaded object is not a tensor " << std::endl;
```

I think we should just throw here, and we can catch all exceptions in `main`.